odl/tomo/geometry/parallel.py

```diff
@@ -1145,31 +1146,33 @@ def parallel_beam_geometry(space, angles=None, det_shape=None):
     # period of twice the inter-node distance.
     min_side = min(space.partition.cell_sides[:2])
     omega = np.pi / min_side
+    num_px_horiz = int(np.ceil(2 * rho * omega / np.pi))
 
-    if det_shape is None:
-        if space.ndim == 2:
-            det_shape = int(2 * rho * omega / np.pi) + 1
-        elif space.ndim == 3:
-            width = int(2 * rho * omega / np.pi) + 1
-            height = space.shape[2]
-            det_shape = [width, height]
+    if space.ndim == 2:
+        det_min_pt = -rho
+        det_max_pt = rho
+        if det_shape is None:
+            det_shape = num_px_horiz
+    elif space.ndim == 3:
+        num_px_vert = space.shape[2]
+        min_h = space.domain.min_pt[2]
+        max_h = space.domain.max_pt[2]
+        det_min_pt = [-rho, min_h]
+        det_max_pt = [rho, max_h]
+        if det_shape is None:
+            det_shape = [num_px_horiz, num_px_vert]
 
-    if angles is None:
-        angles = int(omega * rho) + 1
+    if num_angles is None:
+        num_angles = int(np.ceil(omega * rho))
 
-    # Define angles
-    angle_partition = uniform_partition(0, np.pi, angles)
+    angle_partition = nonuniform_partition(
```

Comment:
An argument for this variant is the principle of least surprise. Given that I will write something like "The angles will be equally spaced between 0 and pi (not including pi).", users will expect that `geometry.angles` will start at exactly 0 and end at pi - epsilon.
And since this call is not visible to the user, there's no weirdness exposed.